src/main/java/com/google/cloud/spanner/r2dbc/SpannerStatement.java

```
@@ -87,15 +87,31 @@ public Statement bindNull(int i, Class<?> type) {
 
   @Override
   public Publisher<? extends Result> execute() {
-    Flux<PartialResultSet> result
-        = this.client.executeStreamingSql(this.session, this.transaction, this.sql);
-
-    return Mono
-        .just(new SpannerResult(
-            Flux.create(sink -> result
-                .subscribe(new ConvertingFluxAdapter(sink, new PartialResultRowExtractor()))),
-            result.next().map(partialResultSet -> partialResultSet.hasStats()
-                ? Math.toIntExact(partialResultSet.getStats().getRowCountExact())
-                : 0)));
+    PartialResultRowExtractor partialResultRowExtractor = new PartialResultRowExtractor();
+
+    return this.client.executeStreamingSql(this.session, this.transaction, this.sql)
+        .switchOnFirst((signal, flux) -> {
+          if (signal.hasError()) {
+            return flux.then(Mono.empty());
```

We should propagate the error:
`Mono.error(signal.getThrowable())`

---

Done.